src/datasources/data-frame/components/DataFrameQueryEditorCommon.tsx

```diff
@@ -0,0 +1,60 @@
+import { QueryEditorProps, CoreApp, SelectableValue } from "@grafana/data";
+import { LoadOptionsCallback } from "@grafana/ui";
+import { getWorkspaceName, getVariableOptions } from "core/utils";
+import _ from "lodash";
+import { DataFrameDataSource } from "../DataFrameDataSource";
+import { DataFrameQuery, DataFrameQueryType, ValidDataFrameQuery } from "../types";
+
+export type Props = QueryEditorProps<DataFrameDataSource, DataFrameQuery>;
+
+export class DataFrameQueryEditorCommon {
+  readonly datasource: DataFrameDataSource;
+  readonly onChange: (value: DataFrameQuery) => void;
+  readonly query: ValidDataFrameQuery;
+  readonly onRunQuery: () => false | void;
+  readonly handleError: (error: Error) => void;
+
+  constructor(readonly props: Props, readonly errorHandler: (error: Error) => void) {
+    this.datasource = props.datasource;
+    this.onChange = props.onChange;
+    this.query = this.datasource.processQuery(props.query);
```

Comment:
I think because the DataFrameVariableQueryEditor uses this class, it will apply the query defaults to the variable model: https://github.com/ni/systemlink-grafana-plugins/blob/1dd516df02da54b709fef8d94131c478081f6bfd/src/datasources/data-frame/types.ts#L17
Which is a little weird because the `tableId` is the only property that matters for the variable query. It probably doesn't affect anything, but it is a little weird. Maybe we could move the `processQuery` call out of the common class.